### PR TITLE
Flexible topography options for hydrostatic losses

### PR DESCRIPTION
**Pull request by yikwill**
# PhysicsNeMo Pull Request

## Description

Make hydrostatic losses more flexible when loading surface geopotential / topography, without breaking existing Hydra configs, and align dataset path specification with new Zarr dataloaders.

- Prefer `dataset_path` + `surface_geopotential_name` (default `z`) and optional `convert_topography_to_meters`
- Keep deprecated `src_directory` / `dataset_name` / `dst_directory` / `data_format` working with `DeprecationWarning`
- Safer named-channel topography indexing; warn if topography looks out of range
- Add `error_histogram(exclude_masked=...)` and `get_tv_error` helpers; RankZero logging when distributed

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

## Dependencies

None.

### CODE REVIEW

File: physicsnemo/metrics/climate/hydrostasy.py [hunk 1]
```diff
@@ -15,13 +15,113 @@
...
+        * ds["constants"].sel(channel_c=surface_geopotential_name).values
+        + surface_geopotential_mean
+    )
+    if convert_topography_to_meters:
+        topography = topography / g0
+
+    topography = torch.tensor(
+        topography[np.newaxis, :, np.newaxis, :, :], dtype=torch.float
+    )
+    logger.info(
```
Comment (daviddpruitt) [inline physicsnemo/metrics/climate/hydrostasy.py]: You probably want to use the logger0 idiom here, otherwise you'll get a flood of messages if you're doing a bigger training job.

Comment (yikwill) [reply physicsnemo/metrics/climate/hydrostasy.py]: Good point, fixed here and also for the topography range logging message.

File: physicsnemo/metrics/climate/hydrostasy.py [hunk 2]
```diff
@@ -440,9 +564,16 @@ def error_histogram(self, prediction, bins, accumulator=None):
         x = self.scale(prediction)
         Tv_avg, Tv_model_avg = self.constraint(x)
         Tv_error = Tv_avg - Tv_model_avg
+
+        valid_mask = torch.ones_like(Tv_error, dtype=torch.bool, device=Tv_error.device)
         # Mask out error in regions below the surface
         if self.topography_masking:
-            Tv_error[x[:, :, 1 : self.num_z_levels, :, :] < self.topography] = 0.0
+            valid_mask[:, :, :Tv_error.shape[2]] = (
+                x[:, :, 1 : self.num_z_levels, :, :] >= self.topography
```
Comment (daviddpruitt) [inline physicsnemo/metrics/climate/hydrostasy.py:587]: The mask works under the assumption that topography has been converted to meters. Its possible that `convert_topography_to_meters` gets set to false with `topography_masking` set to true. The should be a check to make sure the if `topography_masking` is true then `convert_topography_to_meters`  is true

Comment (yikwill) [reply physicsnemo/metrics/climate/hydrostasy.py:587]: You're right. I've added a check that will raise an error at init time if `topography_masking=True` and `convert_topography_to_meters=False`.

**Review by daviddpruitt — APPROVED** [review]
